app/models/manageiq/providers/redhat/infra_manager/refresh/refresher.rb

```diff
@@ -6,20 +6,24 @@ def collect_inventory_for_targets(ems, targets)
       inventory = inventory_from_ovirt(ems)
       raise "Invalid RHEV server ip address." if inventory.api.nil?
 
-      # TODO before iterating over targets it would be good to check whether ExtMgmntSystem is part of it
+      # TODO: before iterating over targets it would be good to check whether ExtMgmntSystem is part of it
       # TODO optimize not to fetch the same objects like clusters for multiple targets
 
       targets_with_data = targets.collect do |target|
         _log.info "Filtering inventory for #{target.class} [#{target.name}] id: [#{target.id}]..."
 
-        case target
-        when Host
-          data,  = Benchmark.realtime_block(:fetch_host_data) { host_targeted_refresh(inventory, target) }
-        when VmOrTemplate
-          data,  = Benchmark.realtime_block(:fetch_vm_data) { vm_targeted_refresh(inventory, target) }
+        # TODO: make sure not to use with api v3
+        if refresher_options.try(:[], :inventory_object_refresh)
```

Comment:
This check determines whether the graph refresh is used or regular refresh. Similar approach we have for other providers.